src/dftbp/dftb/hybridxc.F90

```diff
@@ -1189,6 +1196,61 @@ subroutine getTwoCentralCellLoopCompositeIndex(nAtom0, iAtMN)
   end subroutine getTwoCentralCellLoopCompositeIndex
 
 
+  !> Builds simple composite index for two nested loops over atoms in the central cell.
+  subroutine getiKSiKSPrimeCompositeIndex(env, nS, nK, iKSComposite)
+
+    !> Environment settings
+    type(TEnvironment), intent(in) :: env
+
+    !> Total number of spin channels
+    integer, intent(in) :: nS
+
+    !> Total number of k-points
+    integer, intent(in) :: nK
+
+    !> Composite index for two nested loops over central cell
+    integer, intent(out), allocatable :: iKSComposite(:,:)
+
+    !! Indices of spins/k-points
+    integer :: iS, iK, iKp
+
+    !! Loop counters for global and local composite index
+    integer :: indGlobal, indLocal
+
+    !! Start and end index for MPI parallelization, if applicable
+    integer :: iParallelStart, iParallelEnd
+
+  #:if WITH_MPI
+    call getStartAndEndIndex(nS * nK**2, env%mpi%globalComm%size, env%mpi%globalComm%rank,&
+        & iParallelStart, iParallelEnd)
+  #:else
+    iParallelStart = 1
+    iParallelEnd = nS * nK**2
+  #:endif
+
+    ! Composite index needs to be allocated on all ranks
+    allocate(iKSComposite(3, iParallelEnd - iParallelStart + 1))
+
+    ! Build up composite index iKSComposite for collapsing iKS-iKSPrime summations
+    indGlobal = 1
+    indLocal = 1
+    do iS = 1, nS
+      do iK = 1, nK
+        do iKp = 1, nK
+          if (indGlobal >= iParallelStart .and. indGlobal <= iParallelEnd) then
+            iKSComposite(1, indLocal) = iS
```

Comment:
I wanted to avoid a warning that the compiler has to create a temporary array. Do we have a strategy for these kinds of cases? Probably not performance relevant in the slightest though...

Reply:
We don't seem to

Reply:
I see, in that case I will just keep the current version as there are "only" three indices to set.